ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -597,6 +604,38 @@ func (ipfs *Connector) RepoStat(ctx context.Context) (api.IPFSRepoStat, error) {
 	return stats, nil
 }
 
+// Resolve accepts ipfs or ipns path and resolves it into a cid
+func (ipfs *Connector) Resolve(ctx context.Context, path string) (cid.Cid, error) {
+	validPath, err := gopath.ParsePath(path)
+	if err != nil {
+		logger.Error("could not parse path: " + err.Error())
+		return cid.Undef, err
+	}
+
+	if !strings.HasPrefix(path, "/ipns") && validPath.IsJustAKey() {
+		ci, _, err := gopath.SplitAbsPath(validPath)
+		return ci, err
+	}
+
+	ctx, cancel := context.WithTimeout(ipfs.ctx, ipfs.config.IPFSRequestTimeout)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(ctx, ipfs.config.IPFSRequestTimeout)
```
Due to the addition of the tracing context.